core/src/main/java/feast/core/model/DataSource.java

```diff
@@ -88,27 +89,35 @@ public static DataSource fromProto(DataSourceProto.DataSource spec) {
     DataSource source = new DataSource(spec.getType());
     // Copy source type specific options
     Map<String, String> dataSourceConfigMap = new HashMap<>();
-    switch (spec.getType()) {
-      case BATCH_FILE:
-        dataSourceConfigMap.put("file_url", spec.getFileOptions().getFileUrl());
-        dataSourceConfigMap.put("file_format", spec.getFileOptions().getFileFormat());
-        break;
-      case BATCH_BIGQUERY:
-        dataSourceConfigMap.put("table_ref", spec.getBigqueryOptions().getTableRef());
-        break;
-      case STREAM_KAFKA:
-        dataSourceConfigMap.put("bootstrap_servers", spec.getKafkaOptions().getBootstrapServers());
-        dataSourceConfigMap.put("class_path", spec.getKafkaOptions().getClassPath());
-        dataSourceConfigMap.put("topic", spec.getKafkaOptions().getTopic());
-        break;
-      case STREAM_KINESIS:
-        dataSourceConfigMap.put("class_path", spec.getKinesisOptions().getClassPath());
-        dataSourceConfigMap.put("region", spec.getKinesisOptions().getRegion());
-        dataSourceConfigMap.put("stream_name", spec.getKinesisOptions().getStreamName());
-        break;
-      default:
-        throw new UnsupportedOperationException(
-            String.format("Unsupported Feature Store Type: %s", spec.getType()));
+    JsonFormat.Printer json = JsonFormat.printer();
+    try {
+      switch (spec.getType()) {
+        case BATCH_FILE:
+          dataSourceConfigMap.put("file_url", spec.getFileOptions().getFileUrl());
+          dataSourceConfigMap.put("file_format", json.print(spec.getFileOptions().getFileFormat()));
+          break;
+        case BATCH_BIGQUERY:
+          dataSourceConfigMap.put("table_ref", spec.getBigqueryOptions().getTableRef());
+          break;
+        case STREAM_KAFKA:
+          dataSourceConfigMap.put(
+              "bootstrap_servers", spec.getKafkaOptions().getBootstrapServers());
+          dataSourceConfigMap.put(
+              "message_format", json.print(spec.getKafkaOptions().getMessageFormat()));
+          dataSourceConfigMap.put("topic", spec.getKafkaOptions().getTopic());
+          break;
+        case STREAM_KINESIS:
+          dataSourceConfigMap.put(
+              "record_format", json.print(spec.getKinesisOptions().getRecordFormat()));
+          dataSourceConfigMap.put("region", spec.getKinesisOptions().getRegion());
+          dataSourceConfigMap.put("stream_name", spec.getKinesisOptions().getStreamName());
+          break;
+        default:
+          throw new UnsupportedOperationException(
+              String.format("Unsupported Feature Store Type: %s", spec.getType()));
+      }
+    } catch (InvalidProtocolBufferException e) {
```

Comment:
where exactly this exception happens, do we need to wrap it all in `try`?

Reply:
The exception happens when JSON printer is unable to print the DataFormat Protobuf as JSON. I wrapped the block with try as there are multiple calls to json.print that need a try.